src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js

```diff
@@ -0,0 +1,269 @@
+      statusCounts[executionStatus] = (statusCounts[executionStatus] || 0) + 1
+    }
+  }
+  return statusCounts
+}
+
+const statusCell = workflowObject => {
+  const { succeeded, failed, running, submitted, ...others } = groupCallStatuses(workflowObject.calls)
+  console.log(others)
+
+  return div([
+    table({ style: { margin: '0.5rem' } }, [
+      tbody({}, [
+        submitted ? tr({}, [
+          td(styles.statusDetailCell, [submittedIcon()]),
+          td(['Submitted']),
+          td(styles.statusDetailCell, [submitted])
+        ]) : undefined,
+        running ? tr({}, [
+          td(styles.statusDetailCell, [runningIcon()]),
+          td(['Running']),
+          td(styles.statusDetailCell, [running])
+        ]): undefined,
+        succeeded ? tr({}, [
+          td(styles.statusDetailCell, [successIcon()]),
+          td(['Succeeded']),
+          td(styles.statusDetailCell, [succeeded])
+        ]): undefined,
+        failed ? tr({}, [
+          td(styles.statusDetailCell, [failedIcon()]),
+          td(['Failed']),
+          td(styles.statusDetailCell, [failed])
+        ]): undefined,
+        _.map(other => tr({}, [
+          td(styles.statusDetailCell, [unknownIcon()]),
+          td([other]),
+          td(styles.statusDetailCell, [others[other]])
+        ]), Object.keys(others))
+      ].filter(element => element !== undefined))
+    ])
+
+  ])
+}
+
+const WorkflowDashboard = _.flow(
+  Utils.forwardRefWithName('WorkflowDashboard'),
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  })
+)((props, ref) => {
+  const { namespace, name, submissionId, workflowId, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [workflow, setWorkflow] = useState({})
+  const [showLog, setShowLog] = useState(false)
+  const [failureCopied, setFailureCopied] = useState()
+
+  const signal = Utils.useCancellation()
+
+  /*
+   * Data fetchers
+   */
+
+  useEffect(() => {
+    const initialize = withErrorReporting('Unable to fetch Workflow Details',
+      async () => {
+        // If the workflow is empty, or we need to refresh after 60s:
+        if (_.isEmpty(workflow) || _.includes(workflow.status, ['Running', 'Submitted'])) {
+          if (!_.isEmpty(workflow)) {
+            await Utils.delay(60000)
+          }
+          const includeKey = [
+            'backendLogs',
+            'backendStatus',
+            'end',
+            'executionStatus',
+            'callCaching:hit',
+            'failures',
+            'id',
+            'jobId',
+            'start',
+            'status',
+            'stderr',
+            'stdout',
+            'submission',
+            'submittedFiles:workflow',
+            'subworkflowId',
+            'workflowLog',
+            'workflowName',
+            'workflowRoot'
+          ]
```

Comment:
This should probably be collapsed down to just a few lines, rather than a line per string.

Reply:
And reduced since we only need a handful of these entries

Reply:
TOL one day it would be nice to bring the "constant time" requests and the "might be 10k or 100k values" requests separately to provide at least some of the information faster.